pkg/dao/dao.go

```diff
@@ -17,407 +17,83 @@
-	}
-	dao.client = etcdClient
-	dao.kapi = client.NewKeysAPI(dao.client)
-	return &dao, nil
-}
-
-// SetRaw - Allows the setting of the value json string to the key in the kvp API.
-func (d *Dao) SetRaw(key string, val string) error {
-	_, err := d.kapi.Set(context.Background(), key, val /*opts*/, nil)
-	return err
-}
-
-// GetRaw - gets a specific json string for a key from the kvp API.
-func (d *Dao) GetRaw(key string) (string, error) {
-	res, err := d.kapi.Get(context.Background(), key /*opts*/, nil)
-	if err != nil {
-		return "", err
-	}
-
-	val := res.Node.Value
-	return val, nil
-}
-
-// BatchGetRaw - Get multiple  types as individual json strings
-// TODO: Streaming interface? Going to need to optimize all this for
-// a full-load catalog response of 10k
-// This is more likely to be paged given current proposal
-// In which case, we need paged Batch gets
-// 2 steps?
-// GET /spec/manifest [/*ordered ids*/]
-// BatchGet(offset, count)?
-func (d *Dao) BatchGetRaw(dir string) (*[]string, error) {
-	log.Debug("Dao::BatchGetRaw")
-
-	var res *client.Response
-	var err error
-
-	opts := &client.GetOptions{Recursive: true}
-	if res, err = d.kapi.Get(context.Background(), dir, opts); err != nil {
-		return nil, err
-	}
-
-	specNodes := res.Node.Nodes
-	specCount := len(specNodes)
-
-	log.Debug("Successfully loaded [ %d ] objects from etcd dir [ %s ]", specCount, dir)
-
-	payloads := make([]string, specCount)
-	for i, node := range specNodes {
-		payloads[i] = node.Value
-	}
-
-	return &payloads, nil
-}
-
-// GetSpec - Retrieve the spec for the kvp API.
-func (d *Dao) GetSpec(id string) (*apb.Spec, error) {
-	spec := &apb.Spec{}
-	if err := d.getObject(specKey(id), spec); err != nil {
-		return nil, err
-	}
-	return spec, nil
-}
-
-// SetSpec - set spec for an id in the kvp API.
-func (d *Dao) SetSpec(id string, spec *apb.Spec) error {
-	return d.setObject(specKey(id), spec)
-}
-
-// DeleteSpec - Delete the spec for a given spec id.
-func (d *Dao) DeleteSpec(specID string) error {
-	log.Debug(fmt.Sprintf("Dao::DeleteSpec-> [ %s ]", specID))
-	_, err := d.kapi.Delete(context.Background(), specKey(specID), nil)
-	return err
-}
-
-// BatchSetSpecs - set specs based on SpecManifest in the kvp API.
-func (d *Dao) BatchSetSpecs(specs apb.SpecManifest) error {
-	for id, spec := range specs {
-		err := d.SetSpec(id, spec)
-		if err != nil {
-			return err
-		}
-	}
-
-	return nil
-}
-
-// BatchGetSpecs - Retrieve all the specs for dir.
-func (d *Dao) BatchGetSpecs(dir string) ([]*apb.Spec, error) {
-	payloads, err := d.BatchGetRaw(dir)
-
-	if client.IsKeyNotFound(err) {
-		return []*apb.Spec{}, nil
-	} else if err != nil {
-		return []*apb.Spec{}, err
-	}
-
-	specs := make([]*apb.Spec, len(*payloads))
-	for i, payload := range *payloads {
-		spec := &apb.Spec{}
-		apb.LoadJSON(payload, spec)
-		specs[i] = spec
-		log.Debug("Batch idx [ %d ] -> [ %s ]", i, spec.ID)
-	}
-
-	return specs, nil
+func NewDao() (Dao, error) {
+	return etcd.NewDao()
```

Comment:
Is the idea we'll have multiples from here?

Reply:
Yep, that was the initial plan/thought process